fah_data_run_name_converter.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+
+""" Renumber all RUN folders of a F@H dataset to correspond to RMSD.
+    For instance, RUN0 will be the lowest in RMSD (Angstrom)
+    and RUN N should be the highest in RMSD.
+    The script accepts the working directory of the data Proj<#>
+    and a .csv with the mapping values in the following format:
+
+        OLD RUN, RMSD, NEW RUN
+"""
+import argparse
+import os
+import re
+
+
+def valid_file(path):
+    """ Function to check the existence of a file.
+        Used in conjunction with argparse to check that the given parameter
+        files exist.
+    """
+    value = str(path)
+    if not os.path.isfile(value):
+        raise argparse.ArgumentTypeError(
+            '\"{}\" does not exist'.format(value) +
+            '(must be in the same directory or specify full path).')
+    return value
+
+
+def valid_dir(path):
+    """ Function to check the existence of a directory.
+        Used in conjunction with argparse to check that the given parameter
+        dataset exist.
+    """
+    value = str(path)
+    if not os.path.exists(path):
+        raise argparse.ArgumentTypeError(
+            '\"{}\" does not exist'.format(value) +
+            '(must be in the same directory or specify full path).')
+    return value
+
+
+class FAHDataRunNameConverter(object):
+
+    """Class for renumbering all RUN folders of a F@H dataset to correspond to RMSD."""
+
+    def __init__(self, working_directory, mapper, dry_run=False):
+        if not os.path.isdir(working_directory):
+            raise OSError(
+                2, 'No such file or directory', '{}'.format(working_directory))
+        self.working_directory = working_directory
+        if not os.path.isfile(mapper):
+            raise OSError(2, 'No such file or directory', '{}'.format(mapper))
+        self.mapper = mapper
+        self.dry_run = dry_run
+
+    def display_mapper_info(self, mapper_dict):
+        """Function that displays mapper input information."""
+        print '{}Mapper Information{}'.format('-' * 6, '-' * 6)
+        format_string = '{:<10}{:<10}{:<10}'
+        print format_string.format('OLD RUN', 'RMSD', 'NEW RUN')
+        for key, value in mapper_dict.iteritems():
+            print format_string.format(key, value[0], value[1])
+        print '-' * 30
+
+    def convert_generator(self, mapper_dict, working_directory_walk):
+        """Function to genenerate conversion values."""
+        for root, _, _ in working_directory_walk:
+            if 'RUN' in root:
+                if 'CLONE' not in root:
```

Comment:
It's not only a matter of reusing code but also a matter of readability. When you have `if is_run_directory(working_dir)` it better shows your intention then having the 2 `if` statements. In addition, having nested `if`s making code harder to read.